From 95c1b1fd6d4ca9486cc39fc3e30378f9f4711c02 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Mon, 5 Dec 2016 17:27:53 -0600
Subject: [PATCH] Camera2: override capture support level

-The capture support level detected for our Camera is
LIMITED_YUV which leads to incorrect picture sizes. Well,
they might be for YUV but our Cameras are capable of
LIMITED_JPEG which corrects the camera issues. GservicesHelper
seems to be a place where this detection can be overriden using
a ContentResolver. I thought it would be simpler to just set
a property.

-CaptureSupportLevel is an enum with these values:
ZSL = 1
LEGACY_JPEG = 2
LIMITED_JPEG = 3
LIMITED_YUV = 4
and you can use -1 for unused. The property should correspond
to one of these values.

Change-Id: I72ad06572ecc6fbc7b016f53d2e8a0f50cbe3b6a
---
 src_pd/com/android/camera/util/GservicesHelper.java | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src_pd/com/android/camera/util/GservicesHelper.java b/src_pd/com/android/camera/util/GservicesHelper.java
index 12ae56a..2d951f4 100644
--- a/src_pd/com/android/camera/util/GservicesHelper.java
+++ b/src_pd/com/android/camera/util/GservicesHelper.java
@@ -18,6 +18,7 @@ package com.android.camera.util;
 
 import android.content.ContentResolver;
 import android.content.Context;
+import android.os.SystemProperties;
 
 public class GservicesHelper {
 
@@ -38,11 +39,11 @@ public class GservicesHelper {
     }
 
     public static int getCaptureSupportLevelOverrideBack(ContentResolver contentResolver) {
-        return -1;
+        return SystemProperties.getInt("persist.camera.cap_sup_back",-1);
     }
 
     public static int getCaptureSupportLevelOverrideFront(ContentResolver contentResolver) {
-        return -1;
+        return SystemProperties.getInt("persist.camera.cap_sup_front",-1); 
     }
 
     public static int getMaxAllowedNativeMemoryMb(ContentResolver contentResolver) {
-- 
1.9.1

